### PR TITLE
Volunteer data update app

### PR DESCRIPTION
**Pull request by gilm123**
image

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -11,17 +11,36 @@
...
+    "lib": [
+      "es2017",
+      "dom"
+    ],
     "skipLibCheck": true,
     "skipDefaultLibCheck": true,
     "baseUrl": ".",
+    "types": [
+      "facebook-js-sdk"
+    ],
```
Comment (gilm123) [inline tsconfig.base.json]:
```suggestion
```

File: apps/vaccines-updates-client/src/components/main-form/index.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import Age from './parts/age';
+import TargetGroups from './parts/target-groups';
+import VaccinesAvailability from './parts/vaccines-availability';
+import WorkingHours from './parts/working-hours';
+import { FormItem } from './form-item';
+import { Button } from 'semantic-ui-react';
+
+const logo = require('./resources/logo.jpeg').default;
+
+const Container = styled.div`
```
Comment (idigra) [inline apps/vaccines-updates-client/src/components/main-form/index.tsx:15]: Can't we pluck it to a scss file?
(If not, at least let's separate it to a tsx file which contains only styling).

File: apps/vaccines-updates-client/src/components/main-form/parts/city/index.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { FormItem } from '../../form-item';
+import { CITIES, CITIES_TYPE } from '@vacgaps/constants';
+import { Dropdown } from 'semantic-ui-react'
+import { useFormData } from '../../../../providers/FormDataProvider';
+
+type Props = {
+    className?: string;
+};
+
+const dropDownOptions = Object.entries(CITIES).map(_ => ({
```
Comment (idigra) [inline apps/vaccines-updates-client/src/components/main-form/parts/city/index.tsx]: Sync with main - CITIES dictionary structure has been changed.

File: apps/vaccines-updates-client/src/components/main-form/parts/target-groups/index.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import { Checkbox } from 'semantic-ui-react'
+import styled from 'styled-components';
+import { TargetGroup } from '@vacgaps/constants'
+import { useFormData } from '../../../../providers/FormDataProvider';
+
+type Props = {
+    className?: string;
+};
+
+const TargetGroupsContainer = styled.div`
```
Comment (idigra) [inline apps/vaccines-updates-client/src/components/main-form/parts/target-groups/index.tsx:12]: Same as above, let's separate styling to another file.

File: apps/vaccines-updates-client/src/components/main-form/parts/working-hours/index.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+`;
+
+const Comp: FunctionComponent<Props> = props => {
+    const { setWorkingHours } = useFormData();
+
+    const [startTime, setStartTime] = useState<number>();
+
+    return (
+        <FormItem className={props.className}>
+            <h3>שעות פעילות</h3>
```
Comment (idigra) [inline apps/vaccines-updates-client/src/components/main-form/parts/working-hours/index.tsx]: I don't remember, have we decided what we do with the startTime/endTime issue?
(The issue is that we need a DateTime to indicate the end of spare availability, so if it's not today we know it, and also to not show uncleaned old reports. The structure of hours:minutes doesn't provide it. Also, I think that startTime and endTime of the branch might appear as part of the branch details and not as a standalone property.

File: libs/constants/src/lib/health-care-services.ts [hunk 6]
```diff
@@ -1,9 +1,6 @@
-export const HEALTH_CARE_SERVICES = {
-  0: 'כללית',
-  1: 'מאוחדת',
-  2: 'מכבי',
-  3: 'לאומית',
+export enum HEALTH_CARE_SERVICE {
```
Comment (idigra) [inline libs/constants/src/lib/health-care-services.ts]: Such changes require verifying the report app to work, and also to verify data correction when working against the live data (where healthcare services are still indicated with "0", "1", etc.).

**Review by idigra — CHANGES_REQUESTED** [review]

File: apps/vaccines-updates-client/src/providers/FormDataProvider.tsx [hunk 7]
```diff
@@ -2,15 +2,14 @@ import React, { createContext, FunctionComponent, useState, useContext, useCallb
...
-export type WorkingHours = { start: Time, end: Time };
 export type FormDataContextProps = {
     healthCareService: HEALTH_CARE_SERVICE;
     city: string;
     address: string;
     minimumAge: number;
     targetGroups: TargetGroup[];
     availableVaccines: number;
-    workingHours: WorkingHours;
+    endingTime: number;
```
Comment (idigra) [inline apps/vaccines-updates-client/src/providers/FormDataProvider.tsx]: It's endTime in the contract that is in the FE side (live it like that for now, as the JSON is already aligned to it).

File: apps/vaccines-updates-client/src/components/main-form/parts/city/index.tsx (hunk 3 above)
Comment (gilm123) [reply apps/vaccines-updates-client/src/components/main-form/parts/city/index.tsx]: its a common practice in react, especially when using styled-components

**Review by idigra — CHANGES_REQUESTED** [review]

File: libs/constants/src/lib/targetGroups.ts [hunk 8]
```diff
@@ -0,0 +1,4 @@
+export enum TargetGroup {
+    Cops = 'שוטרים',
+    Teachers = 'מורים',
+};
```
Comment (idigra) [inline libs/constants/src/lib/targetGroups.ts]: Notice all those warnings appear in the PR view in github, e.g. in this file:
image

File: apps/vaccines-updates-client/src/app/app.spec.tsx [hunk 9]
```diff
@@ -23,6 +23,6 @@ describe('App', () => {
       </BrowserRouter>
     );
 
-    expect(getByText('Welcome to vaccines-reporter!')).toBeTruthy();
+    // expect(getByText('Welcome to vaccines-reporter!')).toBeTruthy();
```
Comment (idigra) [inline apps/vaccines-updates-client/src/app/app.spec.tsx]: Clean?

File: apps/vaccines-updates-client/src/components/main-form/index.tsx (hunk 2 above)
Comment (idigra) [reply apps/vaccines-updates-client/src/components/main-form/index.tsx:15]: Talked offline, this is React's standard.

File: apps/vaccines-updates-client/src/providers/FormDataProvider.tsx [hunk 10]
```diff
@@ -1,17 +1,16 @@
 import React, { createContext, FunctionComponent, useState, useContext, useCallback } from 'react';
-import { Time } from '@vacgaps/interfaces';
-import { HEALTH_CARE_SERVICE, TargetGroup } from '@vacgaps/constants';
+import { TargetGroup } from '@vacgaps/constants';
 
 export type FormDataContextProps = {
-    healthCareService: HEALTH_CARE_SERVICE;
+    healthCareService: string;
```
Comment (idigra) [inline apps/vaccines-updates-client/src/providers/FormDataProvider.tsx]: Merge with VaccinesReport.

**Review by idigra — CHANGES_REQUESTED** [review]

**Pull request merged**